modules/digiTrustIdLoader.js

```diff
@@ -0,0 +1,233 @@
+            _savedId = idResult;
+          } catch (ex) {
+            idResult.success = false;
+          }
+          cb(idResult);
+        },
+        fail: function (statusErr, result) {
+          utils.logError('DigiTrustId API error: ' + statusErr);
+        }
+      }
+
+      callApi(opts);
+    }
+  }
+
+  if (window) {
+    window.DigiTrust = mock;
+  }
+}
+
+/*
+ * Internal implementation to get the Id and trigger callback
+ */
+function getDigiTrustId(data, consentData, syncDelay, callback) {
+  if (!isInitialized()) {
+    if (retries >= MAX_RETRIES) {
+      utils.logInfo('DigiTrust framework timeout. Fallback to API and mock.');
+      initMockDigitrust();
+    } else {
+      // use expanding envelope
+      if (retryId != 0) {
+        clearTimeout(retryId);
+      }
+
+      retryId = setTimeout(function () {
+        getDigiTrustId(data, consentData, syncDelay, callback);
+      }, 100 * (1 + retries++));
+      return;
+    }
+  }
+
+  var executeIdRequest = function (data, callback) {
+    DigiTrust.getUser({ member: 'prebid' }, function (idResult) {
+      var exp = (data && data.storage && data.storage.expires) || 60;
+      var rslt = {
+        data: null,
+        expires: exp
+      };
+      if (idResult && idResult.success && idResult.identity) {
+        if (isMock) {
+          // plug in the mock ID
+        }
+
+        if (isFunc(callback)) {
+          rslt.data = idResult.identity;
+          callback(rslt);
+        }
+      } else {
+        // failed
+        if (isFunc(callback)) {
+          rslt.err = 'Failure getting id';
+          callback(rslt);
+        }
+      }
+    });
+  }
+
+  executeIdRequest(data, callback);
+}
+
+function initializeDigiTrust(config) {
+  var dt = window.DigiTrust;
+  if (dt && !dt.isClient && config != null) {
+    dt.initialize(config.init, config.callback);
+  } else if (dt == null) {
+    // Assume we are already on a delay and DigiTrust is not on page
+    initMockDigitrust();
+  }
+}
+
+var testHook = {};
+
+/**
+ * Exposes the test hook object by attaching to the digitrustIdModule.
+ * This method is called in the unit tests to surface internals.
+ */
+function surfaceTestHook() {
+  digitrustIdModule['_testHook'] = testHook;
+}
+
+// the Id system module
+export const digitrustIdModule =
+{
+  name: 'digitrust',
+  decode: function (idData) {
+    try {
+      return { 'digitrustid': idData };
+    } catch (e) {
+      utils.logError('DigiTrust ID submodule decode error');
+    }
+  },
+  getId: getDigiTrustId,
+  init: initializeDigiTrust,
+  _testInit: surfaceTestHook
+}
+
+/**
+ * Expose the loader module in the global for later retrieval
+ */
+function exposeLoader() {
+  let pbjs = getGlobal();
+  if (!pbjs) {
+    utils.logError('DigiTrustLoader failed to find Prebid global');
+    return;
+  }
+
+  config.setConfig({ digitrustId: true });
+
+  if (pbjs.idsys == null) {
+    pbjs.idsys = {};
+  }
+  pbjs.idsys[digitrustIdModule.name] = digitrustIdModule;
```

Comment:
for the sub-module registration, can you use `attachIdSystem` from the UserId Module, for example:
```
// add import
import {attachIdSystem} from './userId.js';

// add submodule to the userId module before the setConfig for userSync.userIds is set (ideally when the .js file inits)
attachIdSystem(digitrustIdModule);
```

Reply:
Yes, I'm in the process of reworking. I'm going to submit a fresh PR once ready.